app/api/groups/delete_group.go

```diff
@@ -0,0 +1,84 @@
+package groups
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:operation DELETE /groups/{group_id} groups groupDelete
+// ---
+// summary: Delete a group
+// description: >
+//   Removes a group, its parent-child relations and other objects linked to it.
+//
+//
+//   The service deletes `groups_groups`, `attempts`, `results`,
+//   `group_membership_changes`, `group_pending_requests`,
+//   `permissions_granted`, `permissions_generated`, and `filters` linked to the group.
+//   Access rights are updated accordingly too.
+//
+//
+//   Restrictions (otherwise the 'forbidden' error is returned):
+//     * the authenticated user should be a manager with `can_manage` = 'memberships_and_group' on the `{group_id}`,
+//     * the group should not be of type "User",
+//     * the group must be empty (no active subgroups of any type).
```

Comment:
The UI needs to be able to distinguish between these two errors.  
(Probably the group not been empty is more a 400?)

Reply:
fixed (with 404)